kinesis/README.md

```diff
@@ -0,0 +1,105 @@
+# Streaming database changes to Amazon Kinesis using Debezium
+
+[Debezium](http://debezium.io/) is a set of Apache Kafka connectors that allows streaming of change events from multiple databases. Using [Debezium Embedded](http://debezium.io/docs/embedded/) it is possible to stream database changes to an arbitrary destination and not be limited to Kafka broker only.
+
+This demo shows how to stream changes from MySQL database running on a local machine to an Amazon [Kinesis](https://aws.amazon.com/kinesis/data-streams/) stream.
+
+## Pre-requisities
+* Java 8 development environment
+* Local Docker installation to run the source database
+* an Amazon [AWS](https://aws.amazon.com/) account
+* [AWS CLI](https://aws.amazon.com/cli/) client
+
+## How to run
+### Start MySQL source database
+We will start a pre-populated MySQL database that is the same as used by the Debezium [tutorial](http://debezium.io/docs/tutorial/).
+
+```mvn docker:run```
+
+### Initialize Kinesis stream
+We suppose that you have already executed `aws configure` as described in AWS CLI [getting started](https://github.com/aws/aws-cli#getting-started) guide.
+
+### Create the Kinesis stream
+
+```aws kinesis create-stream --stream-name kinesis.inventory.customers --shard-count 1```
+
+You can use an arbitrary number of shards. To keep things simple we capture only one table - `customers` from database `inventory` so we need only one stream.
+If you want to capture multiple tables you need to create the equivalent streams for them too.
+The naming shceme of streams is `<engine_name>.<database_name>.<table_name>` which in our case is `kinesis.inventory.<table_name>`.
+
+### Create a stream iterator
+We will use AWS CLI to read messages.
+
+```
+ITERATOR=$(aws kinesis get-shard-iterator --stream-name kinesis.inventory.customers --shard-id 0 --shard-iterator-type LATEST|jq '.ShardIterator')
+```
+
+### Connect the database to Kinesis
+Start the application that uses Debezium Embedded to get change events from database to Kinesis stream.
+```
+mvn exec:java
+```
+
+### Read events from the stream
+Execute command
+
+```
+aws kinesis get-records --shard-iterator $ITERATOR | jq -r '.Records[].Data' | base64 -d | jq .
```

Comment:
I'm having trouble with executing this line. Specifically, I'm wondering how the _base64_ invocation is working for you; the previous _jq_ call yields an array of lines, for each of which _base64_ must be invoked individually. Here it appears though that the entire array is passed at once, causing _base64_ to fail after the first line. Instead it'd need looping through. I'm curious how this works for you; could it be that _base64_ is more capable on Linux that on OS X?

Reply:
Ok, turns out that _jq_ has built-in base64 support in 1.6:

    aws kinesis get-records --shard-iterator $ITERATOR | jq -r '.Records[].Data | @base64d'